py-rattler/src/version.rs

```diff
@@ -0,0 +1,72 @@
+use crate::PyRattlerError;
+use pyo3::{pyclass, pymethods};
+use rattler_conda_types::Version;
+use std::str::FromStr;
+
+#[pyclass]
+#[repr(transparent)]
+#[derive(Clone)]
+pub struct PyVersion {
+    inner: Version,
+}
+
+impl From<Version> for PyVersion {
+    fn from(value: Version) -> Self {
+        PyVersion { inner: value }
+    }
+}
+
+#[pymethods]
+impl PyVersion {
+    #[new]
+    pub fn __init__(version: &str) -> pyo3::PyResult<Self> {
+        Ok(Version::from_str(version)
+            .map(Into::into)
+            .map_err(PyRattlerError::from)?)
+    }
+
+    /// Returns a string representation of the version.
+    pub fn as_str(&self) -> String {
+        format!("{}", self.inner)
+    }
```

Comment:
Should there also be `__repr__` function?

Reply:
Tbere is one in the `version.py` file, right?